### PR TITLE
MM-34569 - remove step 3 to guests

### PR DESCRIPTION
**Pull request by pablovelezvidal**
#### Summary
Remove the step 3 to guest invited members.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-34569

#### Screenshots
GUEST USER:

Captura de pantalla 2021-04-07 a las 23 47 29

NORMAL INVITED USER:
Captura de pantalla 2021-04-07 a las 23 49 14

### CODE REVIEW

**Review by stevemudie — APPROVED** [review]
Tested and Passed

1 - Invited guest user
2 - Logged in as the guest and only observed the first 2 steps of the onboarding flow as expected
3 - Tested fields and moved along and everything worked as expected

**Comment by stevemudie** [comment]
Also tested Member invites and step 3 is in place as expected.

File: components/next_steps_view/steps.ts [hunk 1]
```diff
@@ -116,13 +116,18 @@ export const getSteps = createSelector(
     (state: GlobalState) => isFirstAdmin(state),
     (currentUser, firstAdmin) => {
         let roles = currentUser.roles;
+        const isGuestUser = isGuest(currentUser);
```
Comment (nickmisasi) [inline components/next_steps_view/steps.ts:119]: Check out this part: https://github.com/mattermost/mattermost-webapp/blob/066a51470cf23935aa63ef2018c2282c184a5dfa/components/next_steps_view/steps.ts#L36

We already have support for which steps should show based on role. I think all you need to do to fix this, is to put `system_user` in the roles array for that step.

**Review by nickmisasi — CHANGES_REQUESTED** [review]

**Comment by AGMETEOR** [comment]
@pablovelezvidal thanks for the PR. The JIRA ticket in the subscription is wrong though.

**Review by AGMETEOR — APPROVED** [review]
Otw, LGTM!

File: components/next_steps_view/steps.ts (hunk 1 above)
Comment (pablovelezvidal) [reply components/next_steps_view/steps.ts]: @nickmisasi by placing only system_user in the roles array, it hides the third step for the admin and enables it only for users. If I placed both, system_admin and system_user in the roles array, it shows it for admin but hides it for normal users. Any ideas why is this happening?

**Review by stevemudie — APPROVED** [review]
Re-tested and passed

The steps displayed are different for admins, members & guests and all worked as expected.

First admin
1. Complete your profile
2. Name your team
3. Set up desktop notifications
4. Invite members to the team

Regular member
1. Complete your profile
2. Set up desktop notifications
3. Invite members to the team

Guest
1. Complete your profile

**Review by nickmisasi — APPROVED** [review]
This works for me. You should update the comments on the page though, since the functionality of the filter has changed.

**Review by AGMETEOR — APPROVED** [review]
LGTM

**Pull request merged**